src/data_scraper/scraper_binance.py

```diff
@@ -12,77 +12,87 @@
+        self.currency_pairs = [
+            f'{self.currency_to_buy}{self.currency_to_sell}',
+            *[f'{currency}{self.currency_to_sell}' for currency in all_currencies if currency != self.currency_to_buy]
+        ]
+        self.col_names = ['Timestamp (ms)', 'Open', 'High', 'Low', 'Close', 'Volume', 'Close time', 'Quote asset volume',
                           'Number of trades', 'Taker buy base asset volume', 'Taker buy quote asset volume', 'Ignore']
-
-        self.dataset_path = f"{config.FOLDER_TO_SAVE}/binance"
-        self.dataset_name = f"{self.currency_to_buy}{self.currency_to_sell}"
+        self.datatypes = [int, float, float, float, float, float, int, float, int, float, float, int]
+        self.dataset_path = f"{config.FOLDER_TO_SAVE}/{self.name}/{self.interval}"
+        self.cache_data = None
 
     def get_data(self, start_time, end_time):
-        """Get Binance API exchange rates for all selected currencies. Operates in UTC timezone.
-        By default, it will try to continue from the latest available data point if the last saved timestamp is ahead
-        of the timestamp specified in the config, to save time instead of loading already available data."""
-
+        """
+        Load the exchange data from Binance. Always scrapes data in production.
+        Tries to load data from disk in development. If it is not possible, scrapes the data and saves in parquet.
+        """
+        if not self.dev_run:  # Always scrape if in production
+            return self.scrape_data(start_time, end_time)
+        else:  # Try loading from disk if in development
+            return self.get_stored_data(start_time, end_time)
+
+    def scrape_data(self, start_time, end_time):
+        """
+        Get Binance API exchange rates for all selected currencies. Operates in UTC timezone.
+        """
         df = pd.DataFrame(columns=self.col_names)
 
         for currency_pair in self.currency_pairs:
-            start_time_datetime = time_helpers.timestamp_to_str(start_time, format="exact_time")
-            end_time_datetime = time_helpers.timestamp_to_str(end_time, format="exact_time")
             klines = self.client.get_exchange_rates(
                 currency_to_buy=self.currency_to_buy, currency_to_sell=self.currency_to_sell,
-                interval=self.interval, start_time=start_time_datetime, end_time=end_time_datetime)
+                interval=self.interval, start_time=start_time, end_time=end_time)
 
             temp_df = pd.DataFrame(klines, columns=self.col_names)
-            temp_df[config.MERGE_DATA_ON] = temp_df['Open time'].apply(partial(
-                time_helpers.timestamp_to_str, format='exact_time'))
+
+            # Setting the right data types to save later as metadata in parquet
+            temp_df = temp_df.astype({col_name: dtype for col_name, dtype in zip(self.col_names, self.datatypes)})
```

Comment:
This is nice! Have you thought about creating a dictionary instead of two arrays when initializing the scrapper though? I think it would be more safe.

Reply:
Good idea, done :)